typing/parmatch.ml

```diff
@@ -249,9 +249,9 @@ let simple_match p1 p2 =
   | Tpat_variant(l1, _, _), Tpat_variant(l2, _, _) ->
       l1 = l2
   | Tpat_constant(c1), Tpat_constant(c2) -> const_compare c1 c2 = 0
-  | Tpat_tuple _, Tpat_tuple _ -> true
   | Tpat_lazy _, Tpat_lazy _ -> true
   | Tpat_record _ , Tpat_record _ -> true
```

Comment:
Note: we could check `Array.length lbl1.lbl_all = Array.length lbl2.lbl_all` here as you do for the coherence check. (Currently the tuples are checked but not the records.) If one of the two pattern is empty (no label), they may match.